src/jsxc.lib.gui.js

```diff
@@ -2117,16 +2119,39 @@ jsxc.gui.window = {
             msg: $(this).val()
          });
 
-         $(this).val('');
+         $(this).css('height', '').val('');
+
+         ev.preventDefault();
       }).focus(function() {
+         if (textinputBlurTimeout) {
+            clearTimeout(textinputBlurTimeout);
+         }
+
          // remove unread flag
          jsxc.gui.readMsg(bid);
+
+         resizeTextarea.call(this);
+      }).blur(function() {
+         var self = $(this);
+
+         textinputBlurTimeout = setTimeout(function() {
+            self.css('height', '');
+         }, 1200);
       }).mouseenter(function() {
          $('#jsxc_windowList').data('isOver', true);
       }).mouseleave(function() {
          $('#jsxc_windowList').data('isOver', false);
       });
 
+      function resizeTextarea() {
+         if (!$(this).data('originalHeight')) {
+            $(this).data('originalHeight', $(this).outerHeight());
+         }
+         if ($(this).outerHeight() < this.scrollHeight) {
+            $(this).height($(this).data('originalHeight') * 1.5);
+         }
+      }
+
```

Comment:
This is a subject for debate :laughing:, but it is absolute valid javascript. As you can see at the top of the outer function, the complete function needs to be rewritten. The reason why I defined this helper function at this place was, that it is not used elsewhere in the code and therefore it is easier to read.